src/server/worker/pipeline/transform/registry.go

```diff
@@ -283,16 +277,18 @@ func (reg *registry) processJobRunning(pj *pendingJob) error {
 			return err
 		}
 	}
-	if err := reg.taskQueue.RunTaskBlock(pachClient.Ctx(), func(master *work.Master) error {
-		if err := pj.withParallelDatums(master.Ctx(), func(ctx context.Context, dit datum.Iterator) error {
-			return reg.processDatums(ctx, pj, master, dit)
+	ctx := pachClient.Ctx()
+	taskDoer := reg.driver.NewTaskDoer(pj.ji.Job.ID)
+	if err := func() error {
+		if err := pj.withParallelDatums(ctx, func(ctx context.Context, dit datum.Iterator) error {
+			return reg.processDatums(ctx, pj, taskDoer, dit)
 		}); err != nil {
 			return err
 		}
-		return pj.withSerialDatums(master.Ctx(), func(ctx context.Context, dit datum.Iterator) error {
-			return reg.processDatums(ctx, pj, master, dit)
+		return pj.withSerialDatums(ctx, func(ctx context.Context, dit datum.Iterator) error {
```

Comment:
Was this supposed to change from the `master.Ctx()` to the `pachClient.Ctx()`

Reply:
Yes, there isn't a master anymore and the renewal is handled in the Do call. I'm not sure what you are asking about a master lock, there is no locking involved here (or needed). Each Do call operates in its own prefix in etcd.